kotlinpoet/src/main/java/com/squareup/kotlinpoet/CodeWriter.kt

```diff
@@ -628,18 +628,18 @@ internal class CodeWriter constructor(
    * there are no other opposing modifiers (like [KModifier.PROTECTED] etc.) supplied by the
    * consumer.
    */
-  private fun shouldEmitPublicModifier(
+  private fun shouldEmitImplicitPublicModifier(
     modifiers: Set<KModifier>,
     implicitModifiers: Set<KModifier>,
   ): Boolean {
-    if (modifiers.contains(KModifier.PUBLIC)) {
-      return true
-    }
-
     if (!implicitModifiers.contains(KModifier.PUBLIC)) {
       return false
     }
 
+    if (modifiers.contains(KModifier.PUBLIC)) {
```

Comment:
In fact, is this check even needed? We return false if `public` is an implicit modifier, or if a different visibility modifier is provided; whether or not `public` is in `modifiers` - the method should return true.

Reply:
Yeah in this implementation, it wasn't.

However, I think this approach is kind of confusing, so I went back and just explicitly filtered public in the calling method and then reverted the name/functionality change here. 

Now shouldEmitPublicModifier is the only way the public modifier will be emitted. If checks both explicit and implicit modifiers. If it returns false, public won't be emitted.

I think this is a bit easier to read and understand.

Reply:
Looks great, thanks for taking the time to make it more readable!